src/StreamSelectLoop.php

```diff
@@ -274,7 +273,13 @@ private function streamSelect(array &$read, array &$write, $timeout)
             return @\stream_select($read, $write, $except, $timeout === null ? null : 0, $timeout);
         }
 
-        $timeout && \usleep($timeout);
+        if ($timeout > 0) {
+            \usleep($timeout);
+        } elseif ($timeout === null) {
+            // wait forever (we only reach this if we're only awaiting signals)
+            // this may be interrupted and return earlier when a signal is received
```

Comment:
Is there a test that covers this behavior?

Reply:
This piece of code is hard to put into a unit test simply because it *works* both with and without these lines: Prior to applying these changes, this would cause busy waiting (essentially a `while(true) { /* noop */ }` loop) whereas now it will sleep between each iteration.

Practically, this can be reproduced by running `examples/04-signals.php` manually. Prior to applying this PR, it works, but causes 100% CPU usage. After applying this PR, it still works, but does not cause any noticeable CPU load anymore.

Reply:
Just tried and that confirmed this works as intended and explained :+1: